### PR TITLE
[Typing][A-29] Add type annotations for `paddle/nn/initializer/constant.py`

### PR DESCRIPTION
**Pull request by DrRyanHuang**
### PR Category

User Experience

### PR Types

Improvements

### Description

类型标注：

- paddle/nn/initializer/constant.py

### Related links
 
- https://github.com/PaddlePaddle/Paddle/issues/65008

@SigureMo @megemini

### CODE REVIEW

**Comment by DrRyanHuang** [comment]
想问下 `__init__` 的返回类型一般是 `None` 还是 `ConstantInitializer` 之类的?
```python
from __future__ import annotations
from ...base.initializer import ConstantInitializer

class Constant(ConstantInitializer):
    def __init__(self, value: float = ...) -> None: ...
```
我看这里是前者:
https://github.com/cattidea/paddlepaddle-stubs/blob/main/paddle-stubs/nn/initializer/constant.pyi

**Comment by SigureMo** [comment]
> 想问下 `__init__` 的返回类型一般是 `None` 还是 `ConstantInitializer` 之类的?

None，`__init__` 不返回值，只做初始化，`__new__` 才是创建对象的方法

File: python/paddle/nn/initializer/constant.py [hunk 1]
```diff
@@ -35,13 +37,17 @@ class ConstantInitializer(Initializer):
...
         assert value is not None
         super().__init__()
         self._value = value
         self._force_cpu = force_cpu
 
-    def forward(self, var, block=None):
+    def forward(
+        self,
+        var: paddle.Tensor,
+        block: framework.Block | paddle.pir.Block | None = None,
```
Comment (SigureMo) [inline python/paddle/nn/initializer/constant.py]:
```suggestion
        block: paddle.pir.Block | None = None,
```

统一下吧，静态图逻辑里不暴露老 IR 的概念，动静统一逻辑里不暴露静态图的概念

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**